text/src/autogluon/text/automm/utils.py

```diff
@@ -503,6 +508,72 @@ def create_model(
     else:
         raise ValueError(f"No available models for {names}")
 
+def save_pretrained_model(
+    model: ModuleList,
+    config: DictConfig,
+    path: str
+) -> DictConfig:
+    '''
+    Saving the pretrained weights for offline deployment. Used by turing "standalone=True" in predictor. 
+
+    Parameters
+    ----------
+    model
+        A modulelist containing all models.
+    config
+        A DictConfig object. The model config should be accessible by "config.model".
+    path
+        The saving path to the pretained weights i.e. config.json for "clip" and "hf_text", 'pkl' for "timm_image"
+    '''
+    for idx, model_name in enumerate(config.model.names):
+        if model_name == "clip" or "hf_text" in model_name:
+            model[idx].model.save_pretrained(os.path.join(path,model_name))
+            model_config = getattr(config.model, model_name)
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+        if model_name == "timm_image":
```

Comment:
It seems that saving a pretrained timm checkpoint in `.save()` is not necessary. We can just `pretrained=False` when calling `create_model` in `.load()`. No model downloading is required for timm.

Reply:
+1. @yiqings need to fix it.

Reply:
Is it because timm models weights are already saved in model.ckpt file?